OAK-11267 - Upgrade Azure SDK V8 to V12 for oak-blob-azure

seropian opened the pull request:
[OAK-11267] Upgrade Azure SDK V8 to V12 for oak-blob-azure
Taken over from [https://github.com/apache/jackrabbit-oak/pull/2137]()

reschke commented on oak-blob-cloud-azure/pom.xml:
```diff
@@ -68,6 +71,13 @@
                             azure-core,
                             azure-identity,
                             azure-json,
+                            azure-xml,
```
Exporting these could cause problems; see the current tickets about oak-segment-azure.

seropian replied:
I have read the referenced jira discussions about microsoft exports, indeed in the older version of oak-segment-azure there were some microsoft exports, however in this oak-blob-cloud-azure PR there is no such exported package. The references to azure packages are only within the embed section and is not exported in anyway. It lives only inside the module's classloader and it is not influencing other parts of the runtime.

reschke commented on .gitignore:
```diff
@@ -12,4 +12,6 @@ atlassian-ide-plugin.xml
\ No newline at end of file
+.DS_Store
+.vscode
+.mvn
```
This seems to be an unrelated change.

seropian replied:
fixed

reschke commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataStore.java:
```diff
@@ -41,11 +42,18 @@ public class AzureDataStore extends AbstractSharedCachingDataStore implements Co
     @Override
     protected AbstractSharedBackend createBackend() {
-        azureBlobStoreBackend = new AzureBlobStoreBackend();
+        if(useAzureSdkV12) {
```
```suggestion
        if (useAzureSdkV12) {
```

seropian replied:
fixed

reschke commented on oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/v8/UtilsV8.java:
```diff
@@ -0,0 +1,167 @@
+        if (retries < 0) {
+            return null;
+        }
+        if (retries == 0) {
```
```suggestion
        else if (retries == 0) {
```

seropian replied:
fixed

reschke commented on oak-blob-cloud-azure/src/test/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureDataRecordAccessProviderTest.java:
```diff
@@ -93,19 +97,19 @@ protected void doDeleteRecord(DataStore ds, DataIdentifier identifier) throws Da
     @Override
     protected long getProviderMinPartSize() {
-        return Math.max(0L, AzureBlobStoreBackend.MIN_MULTIPART_UPLOAD_PART_SIZE);
+        return Math.max(0L, MIN_MULTIPART_UPLOAD_PART_SIZE);
```
would be good to avoid unrelated changes

seropian replied:
Extracting the constants was needed because for a while there are going to be two diverging implementations of the Azure Blob Store (supporting sdk v8 and v12), both using the same constants

seropian replied:
I have moved all constants from the newly created  "Constants" to the already existing AzureConstants to simplify

reschke reviewed (changes requested):
Left a few minor comments.

The main issue IMHO is the new export of a microsoft package.

amit-jain commented:
@seropian What are the changes over the previous PR which I reviewed https://github.com/apache/jackrabbit-oak/pull/2137

seropian commented:
@amit-jain , Essentially on the initial PR:
 1. I have fixed a parameter naming bug (accountName -> containerName) and 
 2.  I have moved some Azure constants from a redundant Constants class to the already existing AzureConstants class.
 3. The rest are the small changes mentioned in this review.

reschke commented:
It seems that SonarQube does not run on PRs from non-Oak repos.

Maybe we should move the PR to an Oak branch?

(feedback from colleagues with more Giit/SonarQube karma appreciated)

reschke committed:
OAK-11267: imported diffs from https://github.com/seropian/jackrabbit-oak/tree/upgrade_azure_sdk_12

reschke commented:
Moved into https://github.com/apache/jackrabbit-oak/pull/2409 in Apache Git repo

lweitzendorf committed:
OAK-11299: Missing Segments With Oak Run Segment Copy (#1892)

* OAK-11299: Missing Segments With Oak Run Segment Copy

* add ArchiveIndexComparator

* use ArchiveIndexComparator in segment migrator

* shuffle archive list before sorting

* ArchiveIndexComparator singleton

---------

Co-authored-by: Lucas Weitzendorf <lweitzendorf@adobe.com>

karollewandowski committed:
Fix a typo in builtin_nodetypes.cnd: @peop -> @prop (#2373)

nfsantos committed:
OAK-11834 - Cleanups to reduce changeset in OAK-11814  (#2411)

rishabhdaim committed:
OAK-11801 : removed Guava's fluent iterable with Apache's (#2408)

reschke committed:
OAK-11267: restore line breaks

reschke committed:
OAK-11267: use SystemProperty Supplier

reschke committed:
Revert "OAK-11801 : removed Guava's fluent iterable with Apache's (#2408)"

This reverts commit 629423a76759ff9cb477972207be1e5538671ee2.

reschke committed:
Revert "OAK-11834 - Cleanups to reduce changeset in OAK-11814  (#2411)"

This reverts commit 388e9d8b57bf20e33a2eb303a6c3fadda18e5bd7.

reschke committed:
Revert "Fix a typo in builtin_nodetypes.cnd: @peop -> @prop (#2373)"

This reverts commit ee62c835c2a14640affcf59caec646823e3497d5.

reschke committed:
Revert "OAK-11299: Missing Segments With Oak Run Segment Copy (#1892)"

This reverts commit 8785c8509ae41142a77e68d214a962f5a4a13168.

reschke committed:
OAK-11267: integrate https://github.com/seropian/jackrabbit-oak/commit/613e8d5a4e6c932c3f1f3fad5563785bc62c5559.diff

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure

seropian committed:
Upgrade azure sdk 12 (#2514)

* OAK-11299: Missing Segments With Oak Run Segment Copy (#1892)

* OAK-11299: Missing Segments With Oak Run Segment Copy

* add ArchiveIndexComparator

* use ArchiveIndexComparator in segment migrator

* shuffle archive list before sorting

* ArchiveIndexComparator singleton

---------

Co-authored-by: Lucas Weitzendorf <lweitzendorf@adobe.com>

* Fix a typo in builtin_nodetypes.cnd: @peop -> @prop (#2373)

* OAK-11834 - Cleanups to reduce changeset in OAK-11814  (#2411)

* OAK-11801 : removed Guava's fluent iterable with Apache's (#2408)

* OAK-117141: oak-standalone: update dependencies (#2400)

done.

* Clarify support of SNS in Oak

Only reading is supported via JCR API

* OAK-11824 lucene analyzer factory should be able to load indirect filters (#2413)

* OAK-9447: Upgrade Mongo java driver to 5.2 (#2226)

* OAK-9447: Upgrade Mongo java driver to 5.2 (#1787)
* Conflicts resolved
* Reordered imports like original classes and fixed oak-run about new mongodb-driver-sync
* Upgraded com.mongodb and org.bson versions in Import-Package
* Restored original formatting
* Restored original import ordering

---------

Co-authored-by: raffaega <raffaega@CI00298583>
Co-authored-by: raffaega <raffaega@host.docker.internal>
Co-authored-by: Raffaele Gambelli <r.gambelli@gmail.com>
Co-authored-by: Raffaele Gambelli <raffaele.gambelli@cegeka.com>

* OAK-9447: cleanup formatting changes
* OAK-9447: Fix build and removed unwanted changes due to conflicts.
* OAK-9447: Removed some warnings and reverted some unwanted changes
* OAK-9447: Some imports ordering and use ping command to check Mongo availability
* OAK-9447: reverted some changes in imports and keep functionality.
* OAK-9447: revert some formatting changes.
* OAK-9447: avoid formatting changes
* OAK-9447: Restore functionality of hasWriteConcern.
* OAK-9447: Some empty lines cleaning and fix ReplicaSetDefaultWriteConcernIT
* OAK-9447: remove empty line
* OAK-9447: Bump to version 5.2.1, since 5.2.0 requires an OSGi annotation import.
* OAK-9447: Fix import-packages.
* OAK-9447: Restored previous annotations, as they are not always enforced.
* OAK-9447: Applied PR suggestions.
* OAK-9447: Add back mongoURI
* OAK-9447: Update MongoFullGcNodeBinSumBsonSizeTest to new Mongo Driver.
* OAK-9447: Solve casting issue in MongoFullGcNodeBinSumBsonSizeTest.
* OAK-9447: Updated Number casting on numeric values returned by Mongo.
This is something that changed in Mongo Java Driver 5.2: https://www.mongodb.com/docs/drivers/java/sync/current/reference/upgrade/#version-5.0-breaking-changes
* OAK-9447: Fix type in MongoFullGcNodeBinSumBsonSizeTest.

---------

Co-authored-by: raffaega <raffaega@CI00298583>
Co-authored-by: raffaega <raffaega@host.docker.internal>
Co-authored-by: Raffaele Gambelli <r.gambelli@gmail.com>
Co-authored-by: Raffaele Gambelli <raffaele.gambelli@cegeka.com>

* OAK-11831: Update Oak trunk to Jackrabbit 2.22.2 (#2417)

* OAK-11831: Update Oak trunk to Jackrabbit 2.22.2 - fix version numbers in composite ITs

* OAK-11825 - Improve thread monitoring in the OutOfBandIndexer (#2416)

* OAK-11774: Remove usage of Guava's Splitter - fix split on regex for Windows (#2420)

* OAK-11834: Add delay to getDefaultWriteConcern to avoid UNKNOWN ClusterType (#2422)

* OAK-11834: Add delay to getDefaultWriteConcern to avoid UNKNOWN ClusterType
* OAK-11834: Cover Sharded clusterTypes too

* OAK-11814 - Improve error handling in ElasticResultRowAsyncIterator to deal with timeouts of both slow Elastic queries and slow readers. (#2410)

* OAK-10790: FullTextBinaryTextExtractor fails to extract text from csv (#2414)

* OAK-10790: FullTextBinaryTextExtractor fails to extract text from csv

* OAK-10790: FullTextBinaryTextExtractor fails to extract text from csv- updated comment (#2424)

* OAK-11837: Update commons-csv version to 1.14.1 (#2423)

* OAK-10674 Use Oak's Bloom filter (#2412)

* OAK-11787 ElasticRegexPropertyIndexTest.regexPropertyWithoutFlattened

* OAK-10674 Use Oak's Bloom filter

* OAK-10674 Use Oak's Bloom filter

* OAK-10674 Use Oak's Bloom filter

* OAK-10674 Use Oak's Bloom filter

* OAK-10674 Use Oak's Bloom filter

* Revert "OAK-10674 Use Oak's Bloom filter (#2412)"

This reverts commit c438a3f96a3b7c148a478b098aa59b7a57650c7d.

* OAK-11840: Release Oak 1.84.0 - Candidate Release Notes (#2428)

* OAK-11835: Expose all Mongo connection pool parameters (#2426)

* OAK-11835: Added OSGi configuration for all Mongo Connection Pool parameters

* OAK-11835: Remove unneeded connection to Mongo and move the logging to the actual connection.

* OAK-11835: Add new properties to the builder

* OAK-11835: Avoid "magic number" in Mongo lease timeout connections.

* OAK-11835: Create MongoClientSettings with all the parameters

* OAK-11835: Add detailed connection details logs

* OAK-11835: Imports cleanup

* OAK-11835: This now should be obsolete.

* OAK-11835: Add default constants for Mongo connection pool.

* OAK-11835: Apply default configuration in getDefaultBuilder

* OAK-11835: Added some test cases

* OAK-11835: Fix default leaseSocketTimeout.

* OAK-11835: Add all parameters to log message

* OAK-11835: Fix lease socket timeout connection

* OAK-11835: Use millis instead of ms in parameter definitions

* OAK-11835: Change all the properties to Millis to keep consistency

* OAK-11839 : added throttler to throttle the document store based on external factor set inside settings collection (#2429)

* OAK-11389: added throttler to throttle the document store based on external factor set inside settings collection - sonar complaints (#2431)

* OAK-11843: Improve diagnostics for NPE in SecureNodeBuilder (#2433)

* OAK-11843: Improve diagnostics for NPE in SecureNodeBuilder (#2434)

Added null checks.

* OAK-11845: Make SecureNodeBuilder thread safe (#2435)

done.

* Revert "OAK-11845: Make SecureNodeBuilder thread safe (#2435)"

This reverts commit bf3f78fb26cb85e849a814a65ff15188e060f2c4.

* OAK-11847 - AzureJournalFile fails to rotate journal (#2438)

* OAK-11847 - AzureJournalFile fails to rotate journal

* OAK-11847 - AzureJournalFile fails to rotate journal

* OAK-11840: Release Oak 1.84.0 - Candidate Release Notes

* [maven-release-plugin] rollback the release of jackrabbit-oak-1.82.0

* Revert "[maven-release-plugin] rollback the release of jackrabbit-oak-1.82.0"

This reverts commit 124381da7546aeaa35af78aa1b2ac89a03df4ac9.

* [maven-release-plugin] prepare release jackrabbit-oak-1.84.0

* [maven-release-plugin] prepare for next development iteration

* OAK-11808: Segment graph may need to be recomputed (#2393)

* OAK-11808: SegmentGraph class

* address review comments

* address further comments

---------

Co-authored-by: Lucas Weitzendorf <lweitzendorf@adobe.com>

* OAK-11850 - Use MutableInt to compute facets to avoid boxing/unboxing between int and Integer (#2441)

* OAK-11849 - Make facet evaluation timeout configurable. (#2440)

* OAK-11842: Copy Bloom Filter implementation from oak-run-commons to oak-commons for re-use (#2432)

* OAK-11842: Copy Bloom Filter implementation from oak-run-commons to oak-commons for re-use

* OAK-11842: Copy Bloom Filter implementation from oak-run-commons to oak-commons for re-use - remove unintended changes

* OAK-11842 Bloom Filter: improve javadocs

---------

Co-authored-by: Thomas Mueller <thomasm@apache.org>

* OAK-11852: Use new Bloom filter implementation in document store (#2443)

* OAK-11852: Use new Bloom filter implementation in document store

* OAK-11842: Copy Bloom Filter implementation from oak-run-commons to oak-commons for re-use (#2432)

* OAK-11842: Copy Bloom Filter implementation from oak-run-commons to oak-commons for re-use

* OAK-11842: Copy Bloom Filter implementation from oak-run-commons to oak-commons for re-use - remove unintended changes

* OAK-11842 Bloom Filter: improve javadocs

---------

Co-authored-by: Thomas Mueller <thomasm@apache.org>

---------

Co-authored-by: Thomas Mueller <thomasm@apache.org>

* OAK-11852: add missing test class (#2444)

* OAK-11855: update oak-blob to use oak-commons bloom filter (#2445)

* OAK-11832 : replaced Guava's TreeTraversal with OAK commons (#2421)

* OAK-11832 : replaced Guava's TreeTraversal with OAK commons

* OAK-11832 : replaced Guava's TreeTraversal with OAK commons

* Update oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/Traverser.java

Co-authored-by: Julian Reschke <reschke@apache.org>

* Update oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/Traverser.java

Co-authored-by: Julian Reschke <reschke@apache.org>

* OAK-11832 : added random data generator test to verify guava & oak travsersal codes.

* Update oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/Traverser.java

Co-authored-by: Julian Reschke <reschke@apache.org>

---------

Co-authored-by: Julian Reschke <reschke@apache.org>

* OAK-11856: oak-run-commons: remove usage of old Oak bloom filter, use oak-commons instead (#2446)

* OAK-10856: oak-run-commons: remove usage of old Oak bloom filter, use oak-commons instead

* OAK-10856: oak-run-commons: remove usage of old Oak bloom filter, use oak-commons instead - undo change to search-elastic

* OAK-11840: Release Oak 1.84.0 - oak-doc* versions

* OAK-11858: Remove remains of Oak bloom filter in oak-run-commons (#2449)

* OAK-11859: update Tomcat dependency to 9.0.108 (#2451)

* OAK-11851 - Log performance metrics of statistical facets and simplify logic (#2442)

* OAK-11832 : replaced Guava's TreeTraversal with OAK Commons - fix Import in testcase

* OAK-11224: remove guava dependencies from POMs - oak-store-composite

* OAK-11846 oak-segment-azure - recovery will recover from .bak (#2452)

* OAK-11846 oak-segment-azure - recovery will recover from .bak

* OAK-11833 : replaced Guava's Traverser with OAK commons (#2447)

* OAK-11833 : replaced Guava's Traverser with OAK commons

* OAK-11833 : throw NPE is root node is null

* OAK-11833 : added NPE test for Guava to verify new behaviour with old one

* OAK-11857 - Improve path handling in oak-segment-azure

* OAK-11866 Support for datastore files without length (#2458)

* OAK-11866 Support for datastore files without length

* OAK-11866 Support for datastore files without length

* OAK-11836 : removed Guava's AbstractIterator (#2456)

* OAK-11862: missing facets for multi-valued fields (#2455)

* OAK-11864 : added new ALL_ORPHANS FullGC mode for deleting only ALL_ORPHANS node i.e. without any Empty Props (#2459)

* OAK-9586: SysViewImportHandler does not support expanded names in sv:name attributes

Fixed.

* OAK-11861 - segment store initialization should run in parallel (#2454)

* OAK-11656: Remove usage of Guava Suppliers.memoize  (#2457)

* OAK-11656: Remove usage of Guava Suppliers.memoize - move commons.function to commons.internal.function (#2463)

* OAK-11832 : moved Traverser to internal.graph package and marked as i… (#2462)

* OAK-11832 : moved Traverser to internal.graph package and marked as internal

* OAK-11832 : exported internal.graph package

* OAK-11869 : removed usage of Guava's hashing with apache commons-codec (#2461)

* OAK-11869 : removed usage of Guava's hashing with apache commons-codec

* OAK-11869 : fixed issues with data from ByteBuffer and storing of long values in byteBuffer

* OAK-11869 : moved code to utils to avoid copy/paste

* OAK-11869 : renamed method from hash to hashMurmur32

* OAK-11870: Allow passing fullGCGeneration parameter to RevisionsCommand (#2465)

* OAK-11300: update package dependencies wrt primitives (#2468)

* OAK-11300: shaded-guava: remove packages that are not used (anymore) (ongoing) - primitives

* Revert "OAK-11300: shaded-guava: remove packages that are not used (anymore) (ongoing) - primitives"

This reverts commit c94f4ca08ac14aa30757d42273c22a3b19751ac0.

* OAK-11300: shaded-guava: remove packages that are not used (anymore) (ongoing) - note that primitives are used by concurrent

* OAK-11300: shaded-guava: remove packages that are not used (anymore) (ongoing) - note that primitives are used by concurrent

* OAK-11878 ClusterNodeStoreLock LOCK_TIMEOUT is too large (#2469)

* OAK-11875: Introduced test case for empty properties/Split documents issue (#2470)

* OAK-11864 : fixed 2 failing unit cases (#2473)

* OAK-11875 : added unit cases for EmptyProps and split document issue (#2472)

* OAK-11300: shaded-guava: remove packages that are not used (anymore) … (#2474)

* OAK-9212 AzureArchiveManage.listArchives() should not delete segments (#2467)

* OAK-9212 do not delete blobs when doing AzureArchiveManagerV8#listArchives and AzureArchiveManager#listArchives

* OAK-9212 remove unused imports

* OAK-9212 mock listing blobs during initialisation

* OAK-9212 upload deleted marker when the archive is being deleted, and use the same marker to exclude the archive when invoking listArchives

* OAK-9212 rename method

* OAK-9212 change field visibility

* OAK-9212 upload deleted marker when the archive is being deleted, and use the same marker to exclude the archive when invoking listArchives (Azure SDK v12)

* OAK-9212 use readOnly when instantiating SegmentArchiveManager

* OAK-9212 changed exported version for org.apache.jackrabbit.oak.segment.spi.persistence.split

* OAK-9212 use constant

* OAK-9212 throw UnsupportedOperationException when in read-only mode

* OAK-9212 throw UnsupportedOperationException when in read-only mode

* OAK-9212 added test testListArchivesInReadOnlyModeWithPartiallyDeletedArchive

* OAK-9212 added test testListArchivesInReadWriteModeWithPartiallyDeletedArchive

---------

Co-authored-by: Miroslav Smiljanic <miroslav@apache.com>

* Revert "OAK-9212 AzureArchiveManage.listArchives() should not delete segments…" (#2479)

This reverts commit 2e63881a959d016157f80f4bab6e381b3130d321.

* OAK-11861 - segment store initialization should run in parallel (#2478)

- move ForkJoinUtils to o.a.j.o.commons.internal.concurrent
- align naming (execute/invoke/submit) in ForkJoinUtils with the naming in Java's Executor API

* OAK-11883: Remove usage of Guava collect.Range (#2476)

* OAK-11881: Remove usage of Guava Maps.map() (#2475)

* OAK-11884: AzureArchiveManage.listArchives() should delete segments only if write acces is allowed (#2480)

* OAK-11885: Remove usage of Guava collect.ListMultiMap (#2481)

* OAK-9897 - SplitPersistence: FileReaper cannot finish cleanup (#665)

* OAK-11887: oak-it: exclude *db artefacts (#2483)

* OAK-11886 : replaced Guavs's Futures.immediateFuture with JDK completableFuture (#2482)

* OAK-11888: Remove usage of Guava ThreadBuilderFactory (#2484)

* OAK-11225: remove unused Guava related imports (ongoing)

* OAK-11224: remove guava dependencies from POMs (ongoing) - oak-security-spi (#2485)

* OAK-11224: remove guava dependencies from POMs (ongoing) - guava now only test dependency in oak-core (#2486)

* OAK-11224: shaded guava is only a test dependency in commons, adjust core-spi to re-add the dependency (#2488)

* OAK-11129 Improve Lucene documentation (#2491)

* OAK-11875 : avoid removing props/docs if they have empty split props (#2489)

* OAK-11875 : avoid removing props/docs if they have empty split props

* OAK-11875 : added split props that caused FullGC to skip the document

* Revert "OAK-11224: shaded guava is only a test dependency in commons, adjust …" (#2492)

This reverts commit 1178a985d2be4eaa16291d1519bd7a52ce08d48f.

* OAK-11889 : replaced Guavas's Futures.successfulAsList with JDK comple… (#2487)

* OAK-11886 : replaced Guavs's Futures.successfulAsList with JDK completableFuture

* OAK-11889 : incorporatd review comments

* OAK-11889 : fixed for sonar issue

* OAK-11890 : removed usage of Guava's Futures.allAsList (#2493)

* OAK-11894 resolve the IdentifierManager lazily (#2494)

* OAK-11894 resolve the IdentifierManager lazily

* OAK-11893 : removed usage of Guava's ListenableFuture (#2495)

* OAK-11891 : added stats for documents that were skipped due to presence of empty split props (#2496)

* OAK-11897 Remove exports from oak-segment-azure (#2499)

* OAK-11900: SplitPersistenceTest: disable three tests on Windows platform (#2500)

* OAK-11896 : removed usage of Guava's SettableFuture from Test cases (#2497)

* OAK-11903 : removed usage of Guava's SettableFuture from Prod and its related test files (#2501)

* OAK-11905: Update Mockito dependency to 5.19.0 (#2503)

* OAK-11906 : removed usage of ListableFutureTask (#2502)

* OAK-11908: Update Easymock dependency to 5.6.0 (#2506)

* OAK-11907 : removed Guava's DirectExecutor with oak-commons implementation (#2505)

* OAK-11909 : removed Guava's newDirectExecutorService with oak-commons implementation (#2507)

* OAK-11902: remove existing blobs in AzureTarWriterTest setup (#2512)

Co-authored-by: Lucas Weitzendorf <lweitzendorf@adobe.com>

* OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure

---------

Co-authored-by: Lucas Weitzendorf <lweitzendorf@gmail.com>
Co-authored-by: Lucas Weitzendorf <lweitzendorf@adobe.com>
Co-authored-by: Karol Lewandowski <karol86@gmail.com>
Co-authored-by: Nuno Santos <nsantos@adobe.com>
Co-authored-by: Rishabh Kumar <rishabhdaim1991@gmail.com>
Co-authored-by: mbaedke <manfred.baedke@gmail.com>
Co-authored-by: Konrad Windszus <kwin@apache.org>
Co-authored-by: Fabrizio Fortino <fabrizio.fortino@gmail.com>
Co-authored-by: José Andrés Cordero Benítez <Joscorbe@users.noreply.github.com>
Co-authored-by: raffaega <raffaega@CI00298583>
Co-authored-by: raffaega <raffaega@host.docker.internal>
Co-authored-by: Raffaele Gambelli <r.gambelli@gmail.com>
Co-authored-by: Raffaele Gambelli <raffaele.gambelli@cegeka.com>
Co-authored-by: Julian Reschke <reschke@apache.org>
Co-authored-by: Julian Reschke <julian.reschke@gmx.de>
Co-authored-by: Mohit Kataria <mkataria@apache.org>
Co-authored-by: Thomas Mueller <thomasm@apache.org>
Co-authored-by: Ieran Bogdan <48093736+ierandra@users.noreply.github.com>
Co-authored-by: Julian Sedding <jsedding@apache.org>
Co-authored-by: horia_poradici <horia_poradici@yahoo.com>
Co-authored-by: Miroslav Smiljanic <smmiroslav@gmail.com>
Co-authored-by: Miroslav Smiljanic <miroslav@apache.com>
Co-authored-by: Jörg Hoh <joerghoh@users.noreply.github.com>

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure
- fixed the property name of azure blob log enabling
- obtained property via SystenPropertySupplier

reschke committed:
test

reschke committed:
Merge pull request #2517 from seropian/upgrade_azure_sdk_12

Upgrade azure sdk 12

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure
- fixed the property name of azure blob log enabling
- obtained property via SystenPropertySupplier

reschke committed:
Merge pull request #2518 from seropian/upgrade_azure_sdk_12

OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure
- fix to disable azure sdk v12 by default in AzureDataStore

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure
- added tests and some fixes

reschke committed:
Merge pull request #2520 from seropian/upgrade_azure_sdk_12

Upgrade azure sdk 12

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure
- added tests

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure
- added tests

reschke committed:
Merge pull request #2521 from seropian/upgrade_azure_sdk_12

Upgrade azure sdk 12

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure
- add tests
- fix sonar reported issues

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure

reschke committed:
Merge pull request #2535 from seropian/upgrade_azure_sdk_12

OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure

seropian committed:
Merge branch 'OAK-11267' into upgrade_azure_sdk_12

reschke committed:
Merge pull request #2536 from seropian/upgrade_azure_sdk_12

Upgrade azure sdk 12

seropian committed:
OAK-11267: Upgrade Azure SDK V8 to V12 for oak-blob-azure

seropian committed:
Merge branch 'OAK-11267' into upgrade_azure_sdk_12
